portfolio/src/main/webapp/script.js

```diff
@@ -28,103 +28,233 @@ function addRandomGreeting() {
 }
 
 //*********************** LOADING SCREEN ***********************
-function setVisible(selector, bool) {
+
+/**
+ * Sets elements to visible or not.
+ * @param {string} id - The id of the HTML element.
+ * @param {bool} bool - true to hide, false to show
+ */
+function setHidden(selector, bool) {
     const el = document.getElementById(selector);
-    el.classList.toggle('hiddenclass', bool);
+    el.classList.toggle('hidden', bool);
 }
 
-function swapLoaderForContent() {
-    setVisible('header', false);
-    setVisible('content', false);
-    setVisible('loaderid', true);
+/**
+ * Gets rid of the loading screen and displays the page
+ */
+function swapLoaderForContent(){
+    setHidden('header', false);
+    setHidden('content', false);
+    setHidden('loaderid', true);
 }
 
+// env variables
+let listenNext = null;
+let retrievedInput = null;
+let match_index = 0;
+let matchingNodes = null;
+
 //*********************** SEARCH BAR FUNCTIONS ***********************
-/**
-  * Swaps the loader for the content and enables search
-  */
+
 window.onload = function(){
-    swapLoaderForContent();
+    swapLoaderForContent(); // displays page after loaded
+
+    // Next button event listener, always active
+    listenNext = document.getElementById("nextbutton");
+    listenNext.addEventListener("click", searchHandler);
+
+
+    //Runs search on enter
     document.getElementById('searchbar').onkeydown = function(e){
         if(e.keyCode == 13){
```

Comment:
event.keyCode is deprecated. Use event.key instead:
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keycode

Reply:
Also, when inserting magic numbers, leave a comment of what that number is or assign it to a descriptive variable name. Ex:
```
const spaceKey = 13;
if (e.keyCode == spaceKey) {
  ...
}
```

Reply:
Will do